nix/route53.nix

```diff
@@ -60,6 +60,16 @@ with lib;
       '';
     };
 
+    deployment.route53.usePrivateIp = mkOption {
```

Comment:
The duplicated PR https://github.com/NixOS/nixops/pull/630/files has two improvements over this one that I think could be incorporated for this PR to be merged:

1) better error handling (double if/else logic)

2) mentions that `Private Hosted DNS zones` needs to be used with this option